test/file01.ts

```diff
@@ -0,0 +1,45 @@
+import { CdnDto } from '@xsite-ui/common.cdn';
+import { buildImageSrc } from '@xsite-ui/common.image';
+import { CrossOffersChart } from '@xsite-ui/common.rapido';
+import { OfferProduct } from '@xsite-ui/fintech.shared-types';
+import { getProductIdentifier, isApplicationEntityDataType } from '@xsite-ui/fintech.utils';
+
+import { BucketProps } from '../types';
+
+export function getOffersCrossChartProps(
+  crossOffersChart: BucketProps['crossOffersChart'],
+  cdnData: CdnDto,
+) {
+  const {
+    title = '',
+    secondTitle = '',
+    subTitle = '',
+    numOfOffers = 3,
+    productListV3 = {},
+  } = crossOffersChart || {};
+
+  if (!isApplicationEntityDataType<CrossOffersChart>(crossOffersChart)) {
+    return null;
+  }
+
+  const { products } = productListV3 as unknown as { products: OfferProduct[] };
+
```

Comment:
:robot: OpenAI

The type assertion `as unknown as { products: OfferProduct[] }` is not safe and can lead to runtime errors. Instead, use a type guard or a more specific type annotation to ensure that `productListV3` has a `products` property of type `OfferProduct[]`.

<!-- This is an auto-generated comment by OpenAI -->